integrated_channels/canvas/client.py

```diff
@@ -265,7 +276,114 @@ def delete_course_completion(self, user_id, payload):  # pylint: disable=unused-
+                    current_assignment = current_assignments.get(integration_id)
+                    if current_assignment:
+                        if parse(current_assignment['updated_at']) < parse(assignment['updated_at']):
+                            assignments_to_delete.append(current_assignment['id'])
+                            current_assignments[integration_id] = {
+                                'id': assignment['id'],
+                                'updated_at': assignment['updated_at']
+                            }
+                        else:
+                            assignments_to_delete.append(assignment['id'])
+                    else:
+                        current_assignments[integration_id] = {
+                            'id': assignment['id'],
+                            'updated_at': assignment['updated_at']
+                        }
+
+                # Canvas pagination headers come back as a string ie:
+                # 'headers': {
+                #     'Link': '<{assignment_url}?page=2&per_page=10>; rel="current",' \
+                #             '<{assignment_url}?page=1&per_page=10>; rel="prev",' \
+                #             '<{assignment_url}?page=1&per_page=10>; rel="first",' \
+                #             '<{assignment_url}?page=2&per_page=10>; rel="last"' \
+                # }
+                # so we have to parse out the linked list of assignment pages
+                assignment_page_results = resp.headers['Link'].split(',')
+                pages = {}
+                for page in assignment_page_results:
+                    page_type = page.split('; rel=')[1].strip('"')
+                    pages[page_type] = page.split(';')[0].strip('<>')
+
+                if pages.get('current') == pages.get('last', None):
+                    more_pages_present = False
+                else:
+                    canvas_assignments_url = pages.get('next')
+                    current_page_count += 1
+
+            assignments_removed = self._bulk_remove_course_assignments(canvas_course.get('id'), assignments_to_delete)
+            num_courses_removed += len(assignments_removed)
+
+        if failures:
+            message = 'Failed to dedup assignments for the following courses: {}. Total assignments removed: {}'.format(
+                failures,
+                num_courses_removed
+            )
+            status_code = 400
+        else:
+            message = 'Removed {} duplicate courses from Canvas.'.format(num_courses_removed)
+            status_code = 200
+
+        return status_code, message
+
     # Private Methods
+    def _bulk_remove_course_assignments(self, course_id, assignments_to_remove):
+        """
+        Placeholder
+        """
+        removed_items = []
+        for assignment_id in assignments_to_remove:
+            try:
+                assignment_url = CanvasAPIClient.course_assignments_endpoint(
```

Comment:
I think it may a bit clearner to also pass the assignment_id into the `course_assignments_endpoint` function and output the url?

Reply:
unless we are using it in other places and can't actually do that

Reply:
yeah we use it in two cases, one to get all assignments and another where we get a specific assignment. It felt silly to have two near identical methods just for one small difference `/<assignment_id>`